File: samples/tab-stage-view/nodejs/server/models/adaptiveCard.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

// Adaptive Card for tab to show in stage view
const adaptiveCardForTabStageView = (baseUrl) => {
    // Ensure baseUrl doesn't have trailing slash
    const cleanBaseUrl = baseUrl.replace(/\/$/, '');
    // Encode the URL for use in deeplink context parameter
    const contentUrl = encodeURIComponent(cleanBaseUrl + '/content');
    const websiteUrl = encodeURIComponent(cleanBaseUrl + '/content');
    const stageViewContext = encodeURIComponent(JSON.stringify({
        contentUrl: cleanBaseUrl + '/content',
        websiteUrl: cleanBaseUrl + '/content',
        name: 'DemoStageView'
    }));

    return {
        $schema: 'http://adaptivecards.io/schemas/adaptive-card.json',
        body: [
            {
                type: 'TextBlock',
                size: 'Medium',
                weight: 'Bolder',
                text: 'Click the button to open the url in tab stage view'
            },
            {
                type: 'ActionSet',
                actions: [
                    {
                        type: "Action.Submit",
                        title: "View via card",
                        data: {
                            msteams: {
                                type: "invoke",
                                value: {
                                    type: "tab/tabInfoAction",
                                    tabInfo: {
                                        contentUrl: cleanBaseUrl + "/content",
                                        websiteUrl: cleanBaseUrl + "/content",
                                        name: "Stage view",
                                        entityId: "entityId"
                                    }
                                }
                            }
                        }
                    },
                    {
                        type: "Action.OpenUrl",
                        title: "View via Deep Link",
                        url: `https://teams.microsoft.com/l/stage/${process.env.TeamsAppId}/0?context=${stageViewContext}`
                    }
                ]
            }
        ],
        type: 'AdaptiveCard',
        version: '1.4'
    };
};

module.exports = {
    adaptiveCardForTabStageView,
};

File: samples/tab-stage-view/python/models/adaptive_card.py
import os

def adaptive_card_for_tab_stage_view(base_url_for_open_url: str) -> dict:
    base_url = os.getenv("BaseUrl", "")
    teams_app_id = os.getenv("TeamsAppId", "")

    return {
        "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
        "type": "AdaptiveCard",
        "version": "1.4",
        "body": [
            {
                "type": "TextBlock",
                "size": "Medium",
                "weight": "Bolder",
                "text": "Click the button to open the url in tab stage view"
            },
            {
                "type": "ActionSet",
                "actions": [
                    {
                        "type": "Action.Submit",
                        "title": "View via card",
                        "data": {
                            "msteams": {
                                "type": "invoke",
                                "value": {
                                    "type": "tab/tabInfoAction",
                                    "tabInfo": {
                                        "contentUrl": f"{base_url}/content",
                                        "websiteUrl": f"{base_url}/content",
                                        "name": "Stage view",
                                        "entityId": "stageViewTask"
                                    }
                                }
                            }
                        }
                    },
                    {
                        "type": "Action.OpenUrl",
                        "title": "View via Deep Link",
                        "url": f"https://teams.microsoft.com/l/stage/{teams_app_id}/0?context=%7B%22contentUrl%22%3A%22https%3A%2F%2F{base_url_for_open_url}%2Fcontent%22%2C%22websiteUrl%22%3A%22https%3A%2F%2F{base_url_for_open_url}%2Fcontent%22%2C%22name%22%3A%22DemoStageView%22%7D"
                    }
                ]
            }
        ]
    }
